htdocs/core/lib/tax.lib.php

```diff
@@ -438,19 +438,7 @@ function tax_by_thirdparty($type, $db, $y, $date_start, $date_end, $modetax, $di
 		$sql .= " LEFT JOIN ".MAIN_DB_PREFIX."product as p on d.fk_product = p.rowid";
 		$sql .= " WHERE f.entity IN (".getEntity($invoicetable).")";
 		$sql .= " AND f.fk_statut in (1,2)"; // Paid (partially or completely)
-		if ($direction == 'buy') {
-			if (getDolGlobalString('FACTURE_SUPPLIER_DEPOSITS_ARE_JUST_PAYMENTS')) {
-				$sql .= " AND f.type IN (0,1,2,5)";
-			} else {
-				$sql .= " AND f.type IN (0,1,2,3,5)";
-			}
-		} else {
-			if (getDolGlobalString('FACTURE_DEPOSITS_ARE_JUST_PAYMENTS')) {
-				$sql .= " AND f.type IN (0,1,2,5)";
-			} else {
-				$sql .= " AND f.type IN (0,1,2,3,5)";
-			}
-		}
+		$sql .= " AND f.type IN (0,1,2,3,5)";
```

Comment:
The down/deposit invoices were excluded when using the option _ARE_JUST_PAYMENTS because when we calculate vat on invoice generation, we need to exclude them (since they are no more invoice when hidden option is on).
So you can enable the filter for type = 3 like you did, but only if vat calculation is done on payment and not on invoice.